storage/shard.go

```diff
@@ -305,6 +326,25 @@ func (s *dbShard) Flush(ps persist.Persister) error {
 // a given element. Ignore for now due to lack of proper API.
 // TODO(xichen): Propagate ticking stats back up.
 func (s *dbShard) Tick(ctx context.Context) error {
+	var (
+		numberOfLoadedSegments   int
+		numberOfUnloadedSegments int
+	)
+	if err := s.forEachSegment(func(segment sealedFlushingSegment) error {
+		if segment.LoadedStatus() == segmentUnloaded {
+			numberOfUnloadedSegments++
+		} else {
+			numberOfLoadedSegments++
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	s.metrics.numberOfLoadedSegments.Update(float64(numberOfLoadedSegments))
+	s.metrics.numberOfUnloadedSegments.Update(float64(numberOfUnloadedSegments))
+	s.metrics.numberOfImmutableSegments.Update(float64(s.sealedByMaxTimeAsc.Length))
```

Comment:
This requires RLocking the shard. Is it possible to get this information from within an existing RLock in the ticking path to avoid another RLock overhead?